### DISCUSSION TITLE
Scenes with multiple files, does Clean remove "scenes" or "files"?

### DISCUSSION THREAD

**Discussion by camdurino** [stashapp/stash#5466] [Q&A] [answered]
Just want to confirm if "clean" would only remove the duplicate file and not the scene if another file exists. e.g. `Scene` has afiles `A:\scene_a.mp4` and `Z:\scene_z.mp4`, but I've deleted the files in `Z:\`.

The verbiage in these places (quoted) isn't super clear to me. Running the dry run only flags `scene_z` but I'm scared to clean if it will delete all scenes & meta for a scene where another file still exists. Similar discussion here #5175

Appreciate any clarification!

The text in the "Tasks" tab says: _(emphasis "file")_

> Check for missing files and remove them from the database. This is a destructive action.

But in the docs/help-icon it says: _(emphasis "scene")_

> This task will walk through your configured media directories and remove any scene from the database that can no longer be found. It will also remove generated files for scenes that subsequently no longer exist.

Does anyone else find this to be unclear, or is it just me? Enough for an issue/pr?

**Comment by stg-annon** [comment]
Clean removes scenes that are missing files, it does not remove duplicates

Say you move all your files without scanning them in again and run clean this would delete all you scenes because stash would think all the scenes it previously scanned are now missing

The one exception to this is fileless scenes, scenes that were created without a file to begin with

**Reply by camdurino** [reply]
Hmm. So given my example if I were to run Clean, `Scene`, including meta and generated files, would be deleted because `Z:\scene_z.mp4` doesn't exist, but the file itself `A:\scene_a.mp4` would remain, correct?

It sounds like Clean is not what I'm looking for. Is there any way to remove the file entries for files that have been deleted?

**Reply by stg-annon** [reply, marked as answer]
yes if there are two files attached to a scene if at least one exists the scene will remain

**Reply by camdurino** [reply]
Thank you!
